### PR TITLE
Di 1027 create hailsql

### PR DESCRIPTION
**Pull request by YuAvillachLab**
Code is written by Gabe, working in a server, I'll review and merge it into master now.

### CODE REVIEW

File: IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/QueryController.java [hunk 1]
```diff
@@ -306,12 +306,14 @@ private void validateWhereClause(Resource resource, Entity field,
 			throw new QueryException(
 					"Predicate is not supported by the resource");
 		}
+
 		// Does the predicate support the entity?
-		if ((!predicate.getDataTypes().isEmpty())
+		/*if ((!predicate.getDataTypes().isEmpty())
```
Comment (calypsomatic) [inline IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/QueryController.java]: This won't be a problem anymore?

File: IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/IRCTApplication.java [hunk 2]
```diff
@@ -60,7 +61,9 @@
 	private Map<String, Resource> resources;
 	private Map<String, IRCTJoin> supportedJoinTypes;
 	private Map<ResultDataType, List<DataConverterImplementation>> resultDataConverters;
-	
+
+	// keep objectMapper final
+	public static final ObjectMapper objectMapper = new ObjectMapper();
```
Comment (calypsomatic) [inline IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/IRCTApplication.java]: Good idea!

File: IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/QueryController.java (hunk 1 above)
Comment (YuAvillachLab) [reply IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/QueryController.java]: Hm... never seen any query is using predicate with Entity style before, so this change should be fine...

Comment (YuAvillachLab) [reply IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/QueryController.java]: I preferred not changing it

Comment (YuAvillachLab) [reply IRCT-API/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/QueryController.java]: Will change it back and test it

**Pull request merged**